### DISCUSSION TITLE
Run anthias-viewer on something else than a raspberry pi?

### DISCUSSION THREAD

**Discussion by bu6n** [Screenly/Anthias#1749] [Q&A] [answered]
Hi there! I was tasked to evaluate if we could use Anthias for a project. I've been able to launch `docker-compose.dev.yml` successfully. However, is it possible to test the anthias-viewer without a raspberry pi? I was thinking maybe there's a docker image that I can run on macOS or Linux?

**Comment by vpetersson** [comment, marked as answer]
No as things stand, the viewer only works on the raspberry pi.

**Reply by bu6n** [reply]
Thx for the fast reply!

**Comment by nicomiguelino** [comment]
@bu6n, we now support x86 computers/PCs running Debian 12.

**Reply by yurymkomarov** [reply]
@nicomiguelino I guess any DEB-family OS should work (ex, Ubuntu 24 LTS), right?

**Reply by nicomiguelino** [reply]
@yurymkomarov, it's currently not working on a device with Ubuntu 24.04.
For instance, Screenly/Anthias#2206 says that the installation was not successful and stopped at when the Docker is being installed. Please feel free to open up a pull request or share your code snippets just in case you have something in mind. Thanks.

**Comment by wtm256** [comment]
I am very excited to hear about the support for x86. Trying the install on Debian 12 now.

**Reply by nicomiguelino** [reply]
@wtmanley, let us know your experience with the installation and setup process.
If you think that the docs need work or some parts still need to be automated (say via a Bash script), feel free to create an issue or a pull request.

**Reply by wtm256** [reply]
Install went well on a fresh Debian 12 install for me. I did not specify a root password at install so I would be a sudoer by default, which seems to be what the install script was expecting.

The only thing I really ran into was the blinking cursor from the TTY console persisted in the upper left. I put `echo 0 > /sys/class/graphics/fbcon/cursor_blink` in the `/etc/rc.local` and rebooted with success.

Thanks again for your work to get this to x86. I support public school environments who generally have lots of spare low-power PCs on hand that are perfect for simple digital signage. Commercial/paid software options just aren't feasible for many of them for a simple slide deck or URL/image sets.

**Reply by nicomiguelino** [reply]
@wtmanley
* Thanks for sharing your solution for getting rid of the blinking cursor for x86 devices.
* We're glad that Anthias for PC/x86 (especially for the low-powered ones) is gaining traction in schools, among many others who want to get into digital signage for free.

**Reply by nicomiguelino** [reply]
@wtmanley, we just shipped the fix for the blinking cursor problem in x86 devices. Feel free to let us know (or create a GitHub issue) if something comes up.

**Comment by Grimal** [comment]
Not sure if anyone else has tried, but I’ve installed Anthias on one of those inexpensive TV boxes you can find everywhere for like 40 bucks. Mainly because I had one sitting around collecting dust and I wanted to see if I could get it working. This particular unit has an AmLogic S912 with 2GB ram, and I’d already installed Armbian 26.2.1 bookworm which wasn’t any more difficult than burning the image to a MicroSD card, sticking it in the box and powering up while holding the reset button for a few seconds.

The next part was getting Anthias installed. I ran the install, but it failed. The installer script makes the platform determination quite simply; “if this isn’t x86_64, it’s a raspberry pi.” As such a few things will go wrong installing on this type of board. Not sure what the right answer (if any) is for handling platform detection very specifically, for instance my board has “Octopus Planet” in /proc/device-tree/model, another is NEXBOX A1. There’s no telling what is all out there, but theoretically it would not matter too much if the system is running on a supported release of Armbian. So I just went down the path of “if it’s not x86, and it’s not a Raspberry Pi, it’s a generic ARM board.” It's a good start if nothing else.

After making the necessary edits (more or less just skipping Rasbian specific things), I got the installer to complete, pull down the docker images, and complete the install successfully. I ended up using the pi5 docker images. I don’t have accelerated video support and see 3 or 4 frames a second, but just loading web pages works fine.

**Reply by nicomiguelino** [reply]
@Grimal, thank you for sharing your experience. Feel free to share the details of your changes here or via a pull request.

**Comment by vpetersson** [comment]
Just as an update on this: as of the latest releases, there is now a generic ARM 64 docker image. It has been tested on Rock Pi 4, but there is a [board-enablement.md](https://github.com/Screenly/Anthias/blob/master/docs/board-enablement.md) with details on how to enable ore boards.
